File: src/easy/find-first-palindrome.ts
const check = (word: string): boolean => {
    let left = 0;
    let right = word.length - 1;
    while (left < right) {
        if (word[left] !== word[right]) return false;
    }
    return true;
};
export const findFirstPalindrome = (words: string[]): string => {
    for (const word of words) {
        if (check(word)) return word;
    }
    return "";
};

File: src/main/java/easy/FindFirstPalindrome.java
package easy;

public class FindFirstPalindrome {
  public boolean check(String word) {
    int left = 0;
    int right = word.length() - 1;
    while (left < right) {
      if (word.charAt(right) != word.charAt(left)) return false;
      left++;
      right--;
    }

    return true;
  }

  public String solve(String[] words) {
    for (String word : words) {
      if (check(word)) return word;
    }
    return "";
  }
}
